src/version_set.cc

```diff
@@ -307,16 +323,15 @@ void VersionSet::GetObsoleteFiles(ObsoleteFiles* obsolete_files, SequenceNumber
       auto it = column_families_.find(cf_id);
       if (it != column_families_.end()) {
         it->second->DeleteBlobFile(file_number);
-        if (it->second->files_.empty() && obsolete_columns_.find(cf_id) != obsolete_columns_.end()) {
+        if (it->second->MaybeRemove()) {
           column_families_.erase(it);
-          obsolete_columns_.erase(cf_id);
         }
       } else {
         fprintf(stderr, "column %u not found when deleting obsolete file%" PRIu64 "\n", 
           cf_id, file_number);
         abort();        
       }
-      auto now = tuple_it++;
+      auto now = tuple_it++; // is that okay?
```

Comment:
not sure if it is okay either. change it to `tuple_it = obsolete_files->blob_files.erase(tuple_it)` may be better.

Reply:
from https://en.cppreference.com/w/cpp/container/list/splice
`No elements are copied or moved, only the internal pointers of the list nodes are re-pointed.`
Seems it is fine.